### PR TITLE
Used isValid() to check whether geo coordinates have been properly set

### PR DESCRIPTION
**Pull request by alexeylysenko**
IMO isValid() is more accurate way to validate the geo coordinates than comparing longitude and latitude with 0. In addition, expression like double == int may not work as expected.

### CODE REVIEW

File: src/MissionManager/VisualMissionItem.cc [hunk 1]
```diff
@@ -162,11 +162,11 @@ void VisualMissionItem::setMissionVehicleYaw(double vehicleYaw)
 
 void VisualMissionItem::_updateTerrainAltitude(void)
 {
-    if (coordinate().latitude() == 0 && coordinate().longitude() == 0) {
```
Comment (DonLakeFlyer) [inline src/MissionManager/VisualMissionItem.cc]: This is incorrect. lat=0, lon=0 is a valid coordinate.

Comment (DonLakeFlyer) [reply src/MissionManager/VisualMissionItem.cc]: The other thing to know is that at times for various commands ArduPilot (but not PX4) can use lat=0, lon=0 to indicate a lack of position in a command. So you may see other cases of this.

Comment (alexeylysenko) [reply src/MissionManager/VisualMissionItem.cc]: True, but do we really want to check whether long and lat is zero (we want to skip processing "null island"?) Or it's just a wrong way to catch invalid coordinate.

Comment (DonLakeFlyer) [reply src/MissionManager/VisualMissionItem.cc]: In this specific case there was some transient state which was causing it. I don't quite remember the details. Without going back through commit history to try to find those details it's best to leave it alone and assume the semantics are correct. As opposed to changing it and possibly breaking something in a subtle way.

Comment (alexeylysenko) [reply src/MissionManager/VisualMissionItem.cc]: I see, is it worth to add a comment about that to avoid confusion?

Comment (DonLakeFlyer) [reply src/MissionManager/VisualMissionItem.cc]: Other than what is already there not sure what else is known to be added

**Comment by alexeylysenko** [comment]
Ok, thanks for explanation

**Pull request closed without merging**